src/content/en/updates/2018/06/audio-worklet-design-pattern.md

```diff
@@ -0,0 +1,414 @@
+    AudioWorkletNode processing the audio stream by the user-supplied code. It
+    is instantiated in the AudioWorkletGlobalScope when a AudioWorkletNode is
+    constructed. An AudioWorkletNode can have one matching
+    AudioWorkletProcessor.
+
+## Design Patterns
+
+### Using AudioWorklet with WebAssembly
+
+  - [Example code](https://googlechromelabs.github.io/web-audio-samples/audio-worklet/design-pattern/awn/)
+
+[WebAssembly](https://developer.mozilla.org/en-US/docs/WebAssembly) is a perfect
+companion for AudioWorkletProcessor. The combination of these two features
+brings a variety of advantages to audio processing on the web, but the two
+biggest benefits are: a) bringing existing C/C++ audio processing code into the
+WebAudio ecosystem and b) avoiding the overhead of JS JIT compilation and
+garbage collection in the audio processing code.
+
+The former is important to developers with an existing investment in audio
+processing code and libraries, but the latter is critical for nearly all users
+of the API. In the world of WebAudio, the timing budget for the stable audio
+stream is quite demanding: it is only 3ms at the sample rate of 44.1Khz. Even a
+slight hiccup in the audio processing code can cause glitches. The developer
+must optimize the code for faster processing, but also minimize the amount of JS
+garbage being generated. Using WebAssembly can be a solution that addresses both
+problems at the same time: it is faster and generates no garbage from the code.
+
+The next section describes how WebAssembly can be used with AudioWorklet and the
+accompanied code example can be found
+[here](https://googlechromelabs.github.io/web-audio-samples/audio-worklet/design-pattern/awn/).
+For the basic tutorial on how to use Emscripten and WebAssembly (especially the
+Emscripten glue code), please take a look at
+[this article](/web/updates/2018/03/emscripting-a-c-library).
+
+#### Setting Up
+
+It all sounds great, but we need a bit of structure to set things up properly.
+The first design question to ask is how and where to instantiate a WebAssembly
+module. After fetching the Emscripten's glue code, there are two paths for the
+module instantiation:
+
+  1. Instantiate a WebAssembly module by loading the glue code into the
+    AudioWorkletGlobalScope via `audioContext.audioWorklet.addModule()`.
+  1. Instantiate a WebAssembly module in the main scope, then transfer the
+    module via the AudioWorkletNode's constructor options.
+
+The decision largely depends on your design and preference, but the idea is that
+the WebAssembly module can generate a WebAssembly instance in the
+AudioWorkletGlobalScope, which becomes an audio processing kernel within an
+AudioWorkletProcessor instance.
+
+<figure>
+  <img
+      class="diagram"
+      src="/web/updates/images/2018/06/aw-design-pattern-1.png">
+  <figcaption>
+    WebAssembly module instantiation pattern A: Using <code>.addModule()</code>
```

Comment:
Your list in lines 99-101 uses numbers, right? Maybe pattern A should be pattern 1. Or change the list to use letters.